packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -264,12 +272,50 @@ export const useSliderState = (state: SliderState) => {
     disposables.current = [];
   });
 
-  const valuePercent = getPercent(renderedPosition!, min, max);
-
   // TODO: Awaiting animation time from design spec.
   const animationTime = '0.1s';
 
-  const originPercent = origin ? getPercent(origin, min, max) : 0;
+  const valuePercent = getPercent(renderedPosition!, min, max);
+
+  const originPercent = React.useMemo(() => {
+    return origin ? getPercent(origin, min, max) : 0;
+  }, [max, min, origin]);
+
+  const markValues = React.useMemo((): number[] => {
+    const valueArray: number[] = [];
+
+    // 1. We receive a boolean: mark for every step.
+    if (typeof marks === 'boolean' && marks === true) {
+      for (let i = 0; i < (max - min) / step + 1; i++) {
+        valueArray.push(getPercent(min + step * i, min, max));
+      }
+    } else if (Array.isArray(marks) && marks.length > 0) {
+      // 2. We receive an array with numbers: mark for every value in array.
+      return marks.map(marksItem => getPercent(min + marksItem, min, max));
+    }
+
+    return valueArray;
+  }, [marks, max, min, step]);
+
+  /**
+   * Gets the current percentage position for the marks.
```

Comment:
nit: Since you are using `React.useMemo` now this should probably be:

```suggestion
   * Current percentage position for the marks.
```